src/Fable.Transforms/Replacements.fs

```diff
@@ -469,6 +469,7 @@ let rec equals (com: ICompiler) ctx r equal (left: Expr) (right: Expr) =
         Helper.LibCall(com, modName, "equals", Boolean, [left; right], ?loc=r) |> is equal
     | Builtin (BclGuid|BclTimeSpan|BclTimeOnly)
     | Boolean | Char | String | Number _ | MetaType ->
+        // Helper.LibCall(com, "Util", "physicalEquality", Boolean, [left; right], ?loc=r) |> is equal
```

Comment:
@alfonsogarciacaro I thought the `physicalEquality` wrapper was needed, to allow equality tests like `111 === 222`, which the TS compiler frowns upon, if not wrapped in a call. Perhaps we should leave it.

Reply:
It's tricky, I went back to the equality operator to enable the optimization included in this PR to reduce equality checks at compile time. It seems this optimization is what prevents Typescript from complaining, but we can also "fix" the test by making the variables mutable and adding a potential side effect between the declaration and the assertion to prevent Fable from inlining them... The optimizer has become so good that now we need to fight to have "proper" tests 😂

Reply:
@alfonsogarciacaro We can fix the tests to avoid it, but we can't fix user code.
So if the user does `11 = 22`, it's not going to compile for TypeScript.

Reply:
Sorry, by "fixing" I meant if we want to test the actual runtime equality. After this PR, if we end up with 11 = 22, either by user definition or after inlining, this will be reduced to "false" (and the if/else branch potentially discarded, see operationReduction transform in this PR) so Typescript won't get to see the equality and it should be fine.

Reply:
I see, that's ok then, thanks.